src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java

```diff
@@ -259,6 +260,18 @@ public void apply(Project project) {
             minecraftClassesArtifact = createArtifacts.map(task -> project.files(task.getCompiledArtifact()));
         }
 
+        var supplyDevLogin = project.provider(() -> extension.getRuns().stream().anyMatch(model -> model.getDevLogin().getOrElse(false)));
+
+        // Create a configuration to resolve DevLogin
+        var devLoginConfig = project.getConfigurations().create("devLoginConfig", spec -> {
+            spec.setDescription("This configuration is used to inject DevLogin into the runtime classpath of modding source sets");
+            // We only add DevLogin as a default dependency to allow people to overwrite it
+            spec.defaultDependencies(set -> {
+                // The dependency is added lazily to avoid adding it if no runs request it
```

Comment:
Would it not be easier to always add it to the runtime CP? And is that a good idea or not?

Reply:
Easier? sure. Better? Not really, it would add more jars to waste time on searching classes in.

Reply:
How many classes are we even talking about?

Reply:
At least 19 not including inner classes. But even then, I prefer avoiding needless downloads as much as possible, and I think the arguably negligeable amount of code needed to make it conditional is worth it.